cmd/server-rlimit.go

```diff
@@ -16,9 +16,33 @@
 
 package cmd
 
-import "github.com/minio/minio/pkg/sys"
+import (
+	"io/ioutil"
+	"runtime"
+	"runtime/debug"
+	"strconv"
+	"strings"
+
+	"github.com/minio/minio/pkg/sys"
+)
 
 func setMaxResources() (err error) {
+	// Set the Go runtime max threads threshold to 90% of kernel setting.
+	// Do not return any error when encountered since it is not a crucial task.
+	if runtime.GOOS == "linux" {
```

Comment:
Lets move this to `pkg/sys` ? to have a platform dependent code in its own selective files? for example we can also use cgroup limits here as well.